packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -1,11 +1,19 @@
 import path from 'path';
 import type * as ts from 'typescript';
 
-interface FileSpecifier {
+type FileSpecifier = {
   from: 'file';
   name: string | string[];
-  path?: string;
-}
+} & (
+  | {
+      path?: string;
+      excludePaths?: undefined;
+    }
+  | {
+      path?: undefined;
+      excludePaths?: string | string[]; // defaults to `["node_modules"]`.
+    }
+);
```

Comment:
I have to say I'm not a fan of the new configuration option - what would the use-case be? I can only imagine if you want to specify a local file but prevent the same type defined in another local from being used - but then, why not just use the already-present `path` property to specify the exact type you want?

Reply:
The main use case I thought of and thus why I included it, is to support yarn 2 and 3. Yarn doesn't use the `node_modules` folder but rather is own `.yarn` path.

We could hard code this in too, but yarn allows it to be changed. We could maybe read the config file...

But what if there's another package manager out there doing its own thing too. When bower was still around it used `bower_components` I believe.

Reply:
An alternative would be to allow the `path` option to be a prefix. In which case specifically `src` will stop it from looking in `node_modules`.

Reply:
Ugh, yarn... Yeah, that makes sense. But I think we actually *need* to read this from the config file anyway if we are to support non-standard locations - the `PackageSpecifier` currently hard-codes `node_modules` as well :( So doing that would actually be my preferred solution...

A simpler one would be to just hard-code `node_modules` and `.yarn` and if the user uses custom location, they're on their own :/

Reply:
Or #6861 could actually help - I have written that issue with `PackageSpecifier` in mind, but it could be repurposed for `FileSpecifier` as well - the type just should **not** be found in the map...

Reply:
Sounds promising.